### ISSUE TITLE
Listening for notifications fails

### ISSUE THREAD

**Issue by ralfeus** [atsign-foundation/at_client_sdk#752]
### Describe the bug

When the application listens for notifications at some point the error occurs:

```
WARNING|2022-10-13 11:19:54.136240|Monitor|Heartbeat response not received within expected duration. Heartbeat was sent at 1665659990800, it is now 1665659994136, last heartbeat response was received at 1665659980946. Will close connection, set status stopped, call retryCallback 

INFO|2022-10-13 11:19:54.137650|Monitor|socket.listen onDone called. Will destroy socket, set status stopped, call retryCallback 

INFO|2022-10-13 11:19:54.137699|Monitor|Another closeStopAndRetry operation is in progress 

INFO|2022-10-13 11:20:00.802056|Monitor|status is MonitorStatus.stopped : heartbeat will not be sent 

INFO|2022-10-13 11:20:09.146764|Monitor|starting monitor for @81bored7 with lastNotificationTime: 1665655430193 

INFO|2022-10-13 11:20:10.660778|Monitor|starting monitor for @81bored7 with lastNotificationTime: 1665655430193 

INFO|2022-10-13 11:20:12.381016|Monitor|socket.listen onDone called. Will destroy socket, set status stopped, call retryCallback 

INFO|2022-10-13 11:20:12.381170|Monitor|Retrying start monitor due to error 

INFO|2022-10-13 11:20:12.381300|NotificationServiceImpl|Attempting to retry in less than 15 seconds... Rejected 

INFO|2022-10-13 11:20:27.390790|Monitor|starting monitor for @81bored7 with lastNotificationTime: 1665655430193 

INFO|2022-10-13 11:20:28.148531|Monitor|monitor started for @81bored7 with last notification time: 1665655430193 

Unhandled exception:
NoSuchMethodError: The getter 'isError' was called on null.
Receiver: null
Tried calling: isError
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:38:5)
#1      Monitor._getQueueResponse (package:at_client/src/manager/monitor.dart:302:25)
<asynchronous suspension>
#2      Monitor._authenticateConnection (package:at_client/src/manager/monitor.dart:249:24)
<asynchronous suspension>
#3      Monitor.start (package:at_client/src/manager/monitor.dart:176:7)
<asynchronous suspension>
```
Even though waiting for notifications is wrapped into `try-catch` the exception is still not caught and application is crashed.
A snipped of listening:
```dart
  @override
  void listen() async {
    Logger.root.info("Listening for At notifications");
    final atStream = AtClientManager.getInstance()
        .notificationService
        .subscribe(regex: r'^((?!statsNotification).)*$', shouldDecrypt: true)
      ..handleError((error) {
        Logger.root
            .warning("While waiting for atNotifications an error has occurred: $error");
      });
    final keyRegex =
        RegExp('(@\\w+):([\\w\\.\\-]+)\\.${client.getPreferences()?.namespace}(@\\w+)');
    var retry = true;
    while (retry) {
      try {
        await for (final notification in atStream) {
          if (notification.operation != "update") {
            continue;
          }
        .......
        }
        retry = false;
      } on Exception {
        Logger.root
            .warning("During waiting for AtNotification an unexpected error occurred");
        Logger.root.warning("Restarting listening...");
      }
    }
  }
```

### Steps to reproduce

1. Start listening for notifications with the code similar to one above
2. Wait for some time (can be 4 hours)

### Expected behavior

Listening errors are handled and aren't propagated to the consuming application. 
Or is there are propagated then the issue is documented and can be caught

### Smartphones

- Device: console app
- OS: MacOS

### Were you using an atApplication when the bug was found?

no

**Comment by sitaram-kalluri** [comment]
Found the potential root cause of the issue:
- When the server responses are delayed/timeout, the `monitorResponse` will have null value. In the subsequent line, we check for `isError` which leads to a null pointer exception.
- Approach to fix the delayed monitorResponse from server is to throw `AtTimeoutException` in `getQueueResponse` method which will be handled in `start` method.

**Comment by sitaram-kalluri** [comment]
* Following PR: https://github.com/atsign-foundation/at_client_sdk/pull/763 fixes the issues.
- [x]  Merge to trunk
- [x] Publish at_client package

**Comment by sitaram-kalluri** [comment]
The changes are published in v3.0.42. Hence marking the issue has fixed.
